tests/prop_test.py

```diff
@@ -99,3 +73,53 @@ def test_two_level_transition_propagator():
         for matrix in matrices:
             np.testing.assert_almost_equal(
                 np.matmul(matrix, np.conjugate(matrix.T)), np.eye(2))
+
+
+def test_spatial_superposition_transition_propagator():
+    for n_pulses in [1, 3, 10]:
+        SpatialSuperpositionTransitionPropagator_tester(n_pulses)
+
+
+def SpatialSuperpositionTransitionPropagator_tester(n_pulses):
+    # create initial state ket [1,0,0,...,0] with length 2*n_pulses
+    n_pulses = 3
+    init_state = [1]
+    for i in range(1, 2*n_pulses):
+        init_state.append(0)
+
+    atoms = create_random_thermal_atoms(100, state_kets=init_state)
+    atoms0 = atoms
+    # create Raman beam
+    r_beam = 29.5e-3 / 2  # 1/e^2 beam radius in m
+    wave_vectors = ais.Wavevectors(k1=8.052945514e6, k2=-8.052802263e6)
+    center_Rabi_freq = 2*np.pi/4/13.5e-15

```

Comment:
the unitarity of the time evolution could not be garantued unless the rabi frequency was chosen to be very high. I suspect that has to do with the fact that in the propagate method in prop.py, first the free propagation is done and then that of the internal states. However, if you reverse the process (negative time) they will have slightly different positions.

Reply:
Related to #54.